src/io/argf.cr

```diff
@@ -10,7 +10,7 @@ class IO::ARGF < IO
     @read_from_stdin = false
   end
 
-  def read(slice : Bytes) : Int32
+  def read(slice : Bytes)
```

Comment:
This is actually not necessary anymore because of #10828 which already added a type cast to `Int32`.
That's in fact the proper resolution of these type issues (cf. https://github.com/crystal-lang/crystal/issues/10855#issuecomment-869601026).

We should either remove all restrictions (i.e. revert #10828) or cast all return types to `Int32`.

Reply:
Could also turn these restrictions into `: Int` perhaps?

Reply:
@oprypin IIRC u can't do this for the return type restrictions.

Reply:
I don't see any value in that. Either we remove the potentially erroneous restrictions again or make them true.

Reply:
I'm convinced we should have the restriction and cast to `Int32`. It's a non-breaking change and we had already accepted doing that in #10828. We can easily apply that to all similar methods as well.

Question is only if we do that for 1.1 or 1.2. Technically, it's okay because the restrictions are already in place and the change to make sure they hold under all circumstances is a bug fix.

Reply:
I took the liberty to include the type casts in this PR. Hope that's fine with everyone. Otherwise we can revert e8cf88e.
Doing the same in #10845 btw.

Reply:
Hm I see.
In some way this is the long-term-correct approach.
But in another way, it loses us the ability to use `Int64` in these cases, which might be needed one day.
The current state is indeed consistent with the rest, so that's good.

Reply:
We're set on Int32 in many places, so I don't think this is really much of an issue.